app/assets/javascripts/discourse/app/mixins/composer-video-thumbnail-uppy.js

```diff
@@ -0,0 +1,117 @@
+import Mixin from "@ember/object/mixin";
+import ExtendableUploader from "discourse/mixins/extendable-uploader";
+import UppyS3Multipart from "discourse/mixins/uppy-s3-multipart";
+import Uppy from "@uppy/core";
+import DropTarget from "@uppy/drop-target";
+import XHRUpload from "@uppy/xhr-upload";
+import { warn } from "@ember/debug";
+import I18n from "I18n";
+import getURL from "discourse-common/lib/get-url";
+import { bind } from "discourse-common/utils/decorators";
+import { inject as service } from "@ember/service";
+
+export default Mixin.create(ExtendableUploader, UppyS3Multipart, {
+  dialog: service(),
+  uploadRootPath: "/uploads",
+  uploadTargetBound: false,
+  useUploadPlaceholders: true,
+
+  @bind
+  _handleVideoThumbnailButtonClick(event) {
+    if (!event.target.classList.contains("timestamp-edit-btn")) {
+      return;
+    }
+
+    let video = event.currentTarget.querySelector(
+      ".video-container > video:first-of-type"
+    );
+
+    let video_src = video.getElementsByTagName("source")[0].src;
+    let video_sha1 = video_src
+      .substring(video_src.lastIndexOf("/") + 1)
+      .split(".")[0];
+
+    let canvas = document.createElement("canvas");
+    let ctx = canvas.getContext("2d");
+    let videoHeight, videoWidth;
+    videoHeight = video.videoHeight;
+    videoWidth = video.videoWidth;
+    canvas.width = videoWidth;
+    canvas.height = videoHeight;
+
+    ctx.drawImage(video, 0, 0, videoWidth, videoHeight);
+
+    // upload video thumbnail
+    canvas.toBlob((blob) => {
```

Comment:
I pulled this branch locally, and if the image URL and site domain don't match, I get this error: 

```
Failed to execute 'toBlob' on 'HTMLCanvasElement': Tainted canvases may not be exported.
```

Locally I get this for `localhost` vs `127.0.0.1` but in prod this can happen, too, maybe?

Reply:
Ya, locally it has to be 127.0.0.1. I don't think this would happen in prod. I think this is just related to our ember_cli proxy. We do use a some other canvas exporting related things in core already that work in prod:

```
app/assets/javascripts/discourse/app/lib/update-tab-count.js
49:  newFavicon.href = canvas.toDataURL("image/png");
```